### DISCUSSION TITLE
Embed ETCD to distributed ETCD

### DISCUSSION THREAD

**Discussion by xiaofan-luan** [etcd-io/etcd#15756] [Q&A] [answered]
If there a way to migrate embedded ETCD into a distributed ETCD?

**Comment by jmhbnz** [comment]
There isn't a lot of detail in your question so I will have to make the assumption you are asking about migrating etcd data. 

If that is the case, provided you can interact with the embedded etcd instance via something like `etcdctl` you should be able to follow the snapshot and restore process.

1) Setup your new etcd distributed cluster

2) Export data from old etcd embedded instance via `etcdctl`.

3) Restore data to new etcd distributed members via `etcdutl`.

The snapshot and restore process is documented in more detail here: https://etcd.io/docs/v3.5/op-guide/recovery.

**Reply by xiaofan-luan** [reply, marked as answer]
> There isn't a lot of detail in your question so I will have to make the assumption you are asking about migrating etcd data.
> 
> If that is the case, provided you can interact with the embedded etcd instance via something like `etcdctl` you should be able to follow the snapshot and restore process.
> 
> 1. Setup your new etcd distributed cluster
> 2. Export data from old etcd embedded instance via `etcdctl`.
> 3. Restore data to new etcd distributed members via `etcdutl`.
> 
> The snapshot and restore process is documented in more detail here: https://etcd.io/docs/v3.5/op-guide/recovery.

Great, that's just what I'm looking for. Thanks man
